### DISCUSSION TITLE
How to secure user sensitive files?

### DISCUSSION THREAD

**Discussion by denisgurev** [pocketbase/pocketbase#2263] [Q&A] [answered]
I want to allow my users to upload contracts and other various sensitive documents, but I can't seem to make them private (even with admin level API rules).

Is this a bug or just how it works and can I customize it?

To clarify I want only the uploader to have access to the uploaded files in the same way you can restrict access to the record data.

**Comment by ganigeorgiev** [comment, marked as answer]
Currently all uploaded files are publicly accessible as long as you have the full file url.

Private/protected files are already implemented in the [`develop` branch](https://github.com/pocketbase/pocketbase/tree/develop) and will be available with the next v0.15.0 release as part of https://github.com/pocketbase/pocketbase/issues/215#issuecomment-1497270628.
